src/testing/native-debugging.md

```diff
@@ -0,0 +1,458 @@
+in [Update test Flutter app](#update-test-flutter-app).
+
+{% include docs/debug/debug-flow-android.md %}
+
+### Debug Dart and iOS code using Xcode
+
+To debug iOS code, you need a Flutter app that contains iOS code.
+In this section, you learn to connect two debuggers to your app:
+Flutter via VS Code and Xcode. You need to run both VS Code and Xcode.
+
+These section uses the same example Flutter `url_launcher` app created
+in [Update test Flutter app](#update-test-flutter-app).
+
+{% include docs/debug/debug-flow-ios.md %}
+
+### Debug Dart and macOS code using Xcode
+
+To debug macOS code, you need a Flutter app that contains macOS code.
+In this section, you learn to connect two debuggers to your app:
+Flutter via VS Code and Xcode. You need to run both VS Code and Xcode.
+
+These section uses the same example Flutter `url_launcher` app created
+in [Update test Flutter app](#update-test-flutter-app).
+
+{% include docs/debug/debug-flow-macos.md %}
+
+### Debug Dart and C++ code using Visual Studio
+
+To debug C++ code, you need a Flutter app that contains C++ code.
+In this section, you learn to connect two debuggers to your app:
+Flutter via VS Code and Visual Studio.
+You need to run both VS Code and Visual Studio.
+
+These section uses the same example Flutter `url_launcher` app created
+in [Update test Flutter app](#update-test-flutter-app).
+
+{% include docs/debug/debug-flow-windows.md %}
+
+## Resources
+
+Check out the following resources on debugging Flutter, iOS, and Android:
+
+### Flutter
+
+- [Debugging Flutter apps][]
+- [Flutter inspector][] and the [DevTools][] docs.
+- [Performance profiling][]
+
+[Debugging Flutter apps]: {{site.url}}/testing/debugging
+[Performance profiling]: {{site.url}}/perf/ui-performance
+
+### Android
+
+You can find the following debugging resources on
+[developer.android.com][].
+
+- [Debug your app][]
+- [Android Debug Bridge (adb)][]
+
+### iOS and macOS
+
+You can find the following debugging resources on
+[developer.apple.com][].
+
+- [Debugging][]
+- [Instruments Help][]
+
+### Windows
+
+You can find debugging resources on [Microsoft Learn][].
+
+- [Visual Studio Debugger][]
+- [Learn to debug C++ code using Visual Studio][]
+
+[Android Debug Bridge (adb)]: {{site.android-dev}}/studio/command-line/adb
+[Debug your app]: {{site.android-dev}}/studio/debug
+[Debugging]: {{site.apple-dev}}/support/debugging/
+[developer.android.com]: {{site.android-dev}}
+[developer.apple.com]: {{site.apple-dev}}
+[DevTools]: {{site.url}}/tools/devtools
+[Flutter inspector]: {{site.url}}/tools/devtools/inspector
+[Instruments Help]: https://help.apple.com/instruments/mac/current/
+[Microsoft Learn]: https://learn.microsoft.com/en-us/visualstudio/
+[Visual Studio Debugger]: https://learn.microsoft.com/en-us/visualstudio/debugger/?view=vs-2022
+[Learn to debug C++ code using Visual Studio]: https://learn.microsoft.com/en-us/visualstudio/debugger/getting-started-with-the-debugger-cpp?view=vs-2022
```

Comment:
Microsoft Learn will redirect to the user's preferred language only if you omit the language ID:

```suggestion
[Microsoft Learn]: https://learn.microsoft.com/visualstudio/
[Visual Studio Debugger]: https://learn.microsoft.com/visualstudio/debugger/?view=vs-2022
[Learn to debug C++ code using Visual Studio]: https://learn.microsoft.com/visualstudio/debugger/getting-started-with-the-debugger-cpp?view=vs-2022
```